src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx

```diff
@@ -584,17 +602,52 @@ class AddItemsPage extends Component {
    * @public
    */
   saveAndTransitionToNextStep(formValues, lineItems) {
-    if (_.some(lineItems, item => item.inventoryItem
-      && item.expirationDate !== item.inventoryItem.expirationDate)) {
-      if (_.some(lineItems, item => item.inventoryItem && item.inventoryItem.quantity && item.inventoryItem.quantity !== '0')) {
-        this.confirmInventoryItemExpirationDateUpdate(() =>
-          this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems));
-      } else {
-        this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems);
-      }
-    } else {
-      this.saveRequisitionItemsAndTransitionToNextStep(formValues, lineItems);
+    const expirationDateOrLotChanged = _.some(lineItems, (item) => {
+      const inventoryItem = item?.fetchedInventoryItem?.inventoryItem || item?.inventoryItem;
+      const quantity = item?.fetchedInventoryItem?.quantity || item?.inventoryItem?.quantity;
```

Comment:
@alannadolny please make sure that this `item?.inventoryItem?.quantity` is always populated with the proper quantity on hand from the product availability (and where it is populated). If that isn't correct qoh, then I think there will be a need to fetch inventory item data again when the user changes the expiry date for some row after **re-entering** the page (**not** after adding new row)

Reply:
Yeah, we should always distinguish between quantityAvailable and quantityOnHand.

Reply:
@awalkowiak 
![image](https://github.com/openboxes/openboxes/assets/83239466/25f0d5ca-9e15-47ca-8af2-99d66501a467)
The quantity that we are using on fetching line items is from `StockMovementService` line 777. So, it's quantity on hand. When the inventory items are new, the quantity is null instead of 0, that's the reason it didn't work when I was showing that during the last tech huddle.